### ISSUE TITLE
Work nicely with Camera, File, and Ionic-Keyboard Plugins (on Ionic multi-pages app)

### ISSUE THREAD

**Issue by JamalANasier** [brody2consult/cordova-sqlite-storage#508]
This is not an issue actually. I am writing this as my gratitude for @brodybits and every person behind this plugin. I want to share my experiences with some tips and tricks that help me along the way.
I am using Ionic Framework, the details  are as follow :
- Cordova CLI: 6.1.1
- Gulp version:  CLI version 3.9.1
- Ionic Version: 1.1.0
- Ionic CLI Version: 1.7.14
- Ionic App Lib Version: 0.7.0
- ios-deploy version: 1.8.6 
- ios-sim version: 5.0.8 
- OS: Mac OS X El Capitan
- Node Version: v6.2.0
- Xcode version: Xcode 7.3.1

The pages in total are 13, with 3 of them are only for text and picture without connecting to the SQLite database. The rest (10 pages) are connecting to database and use other plugins functionality such camera and file plugins. I will not write the detail $state because it will be too long to read :). So here we go..

Taking picture and store somewhere with the file plugin, and also store the link to the picture to SQLite :
 First Code : `$scope.takePicture = function () {
        var option = {
            destinationType: Camera.DestinationType.FILE_URI,
            sourceType: Camera.PictureSourceType.CAMERA,
            encodingType: Camera.EncodingType.JPG
        };
        $cordovaCamera.getPicture(option)
            .then(function (imageURI) {
                    console.log('NATIVE_URI==>' + imageURI);
                    var name = imageURI.substr(imageURI.lastIndexOf('/') + 1);
                    var namePath = imageURI.substr(0, imageURI.lastIndexOf('/') + 1);
                    var newName = makeid() + name;
                    $cordovaFile.copyFile(namePath, name, cordova.file.dataDirectory, newName)
                        .then(function (fileEntry) {
                            $rootScope.urlInsert = fileEntry.nativeURL;
                            resolve();
                        }, function (e) {
                            reject();
                        });
                },
                function (error) {
                    console.log('camera error :');
                });
    };`
There is no any of this plugin codes above right ? since there is something i didn't know what happened when the first time i implemented this plugin with file plugin and camera plugin, my app break. So i have to find a work around to be safe. As you can see above my safeplay is on $rootScope.urlinsert. Will see as below :
Second Code (on another controller) :

```
.controller('InputCtrl', function ($scope, $rootScope, $cordovaSQLite, $ionicHistory) {
    $scope.listItems = [];
    $rootScope.urlInsert = $rootScope.urlInsert;
    window.addEventListener('native.keyboardshow', function () {
        document.body.classList.add('keyboard-open');
    });
    $scope.save = function (name, date, place, photo) {
        var query = "INSERT INTO chocolates (name, date, place, photo) VALUES (?, ?, ?, ?)";
        // execute INSERT statement with parameters
        $cordovaSQLite.execute(db, query, [name, date, place, $rootScope.urlInsert]).then(function (result) {
            $scope.statusMessage = "data insert successfully.!";
            delete $rootScope.urlInsert
        }, function (error) {
            $scope.statusMessage = "Error on saving: " + error.message;
        });
    };

    $scope.$on('$ionicView.leave', function () {
        $ionicHistory.clearCache();
    });
})
```

As you can see now $rootScope.urlInsert comes in play, the value from the camera controller passed in to this controller as input value (path where file plugin store the  image). I am a noob so i cannot make any services as workaround i use $rootScope to pass the value between controllers ^_^.  Then i delete this rootScope on success callback, because the value will stay there until new input taken. On the end i call IonicHistory.Clearcache() to see the effect on display pages for the SQlite database values. 

There is some other tricks but i think this is the important one to show that this plugin works well with others such as camera and file plugin. And also with multi-pages app. I cannot show you guys the link yet since the app is still on TestFlight. Owh i also have one page that showing slider by idangerous with the data from SQLite 💃 

So goodluck.. and happy coding (will end here before it gets too long and bored you lol)

**Comment by brody2consult** [comment]
Thanks @JamalANasier! Unfortunately I find your description a bit hard to follow. I wish your "First Code" would have been a proper block like you did for the "Second Code". I think this would make a nice tutorial if you clean it up and make it easier to understand. I would also be interested to use your app as a sample once it is ready.

**Comment by JamalANasier** [comment]
Hehehe sorry buddy, but trust me i already tried twice to make as good as the codes below that one. Maybe someone can edit that 👍 

sure when it's ready, if you approve it i will share the link here

**Comment by amrayoub** [comment]
@JamalANasier  Hi Jamal , i'm currently trying to do the same thing , save the file url of the image in db along with time stamp and different ID for each image , can u please send me the full code along with html , i would really appercaite it

**Comment by brody2consult** [comment]
This kind of discussion should be on Cordova-sqlite-help.
